rstudio/rstudio
# improve handling of errors in render output

### Intent

Addresses https://github.com/rstudio/rstudio/issues/9390.

### Approach

The core of the issue is simple (using `white-space: pre;`), but this PR also fixes an issue where the <kbd>(Output | Issues)</kbd> button was inconsistently shown / not-shown. That was ultimately caused by our code failing to account for output being streamed in in separate chunks, which may or may not be overlapping.

### Automated Tests

N/A

### QA Notes

Given the following R Markdown document:

`````
---
title: "fail"
---

```{r}
f <- function() g()
g <- function() h()
h <- function() rlang::abort("This is an error!")

f()
```
`````

Try to Knit this document multiple times, and confirm that the error output is presented neatly, and the <kbd>(Output | Issues)</kbd> button is always displayed. (In older versions of RStudio, you should see it missing roughly 10% of the time.)

<img width="674" alt="Screen Shot 2021-07-09 at 11 47 40 AM" src="https://user-images.githubusercontent.com/1976582/125123410-790f3000-e0ab-11eb-9f73-83e5a7b2c968.png">

### Checklist
